Source/SwiftLintFramework/Models/LinterCache.swift

```diff
@@ -13,12 +13,31 @@ public enum LinterCacheError: Error {
     case invalidFormat
     case differentVersion
     case differentConfiguration
+    case inconsistentLastRunDate
 }
 
 public final class LinterCache {
     private var cache: [String: Any]
     private let lock = NSLock()
 
+    public var lastRunDate: Date? {
+        get {
+            lock.lock()
```

Comment:
Is it possible to write an unlocker class that would own a NSLock instance and unlock it in its deinit method, like a C++ smart pointer? Having these unlock()s everywhere seems messy and error prone.